code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -24,6 +29,181 @@
 		return TRUE
 	. = ..()
 
+/obj/item/reagent_containers/food/drinks/cans/update_icon()
+	cut_overlays()
+	var/image/fuseoverlay = image('icons/obj/fuses.dmi', icon_state = "fuse_short")
+	switch(fuselength)
+		if(1 to 5)
+			if("x" in can_size_overrides)
+				fuseoverlay.pixel_x = can_size_overrides["x"]
+			if("y" in can_size_overrides)
+				fuseoverlay.pixel_y = can_size_overrides["y"]
```

Comment:
We can ditch these if checks and instead make the list on all cans have something like:
`list("x" = 0, "y" = 0)`
so that the default position is, well, the default one. Then we don't have to use if checks because the overrides will be present on all cans.